bin/database-diagram.py

```diff
@@ -0,0 +1,236 @@
+#!/usr/bin/env python3
+
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) 2008-2019 NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""database-diagram
+
+Create a database diagram of Cylc's database, reading its schema, and adding
+relationships manually.
+
+The console output should include the markdown and the dot file used,
+and a PNG image with the database diagram will be created in the working
+directory, with the name cylc-database.png."""
+
+import logging
+import sqlite3
+import tempfile
+from contextlib import closing
+
+import pygraphviz as pgv
+from cylc.flow.rundb import CylcSuiteDAO
+from eralchemy import render_er, main as eralchemy_main
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+def get_relationships():
+    """These are the database relationships. * means 0..N, and ? {0, 1}.
+
+    Ideally we would have these relationships in the database model, but doing
+    that would require further tests to prevent regressions for users
+    after they update Cylc. So we are hard-coding the relationships here
+    for now.
+
+    Returns:
+        list: with relationships in the eralchemy markdown syntax.
+    """
+    # TODO: remove this once the relationships are in the DB, and then automate
+    return [
+        ["task_states", "*--?", "task_events"],
```

Comment:
Even if these are temporary, using two constants like ``ZERO_TO_N_REL = *--?`` etc. might be more comprehensible for development & ensures consistency.

Reply:
Good idea. Will also use the table name constants, e.g. `CylcSuiteDAO.TABLE_TASK_STATES` :+1: